### PR TITLE
Fix a couple of typos/issues in s3 error handling

### PR DESCRIPTION
**Pull request by oavdeev**
...caught by `S3FailureTest`

### CODE REVIEW

**Review by savingoyal — APPROVED** [review]
lgtm! waiting for tests to pass. @oavdeev can we enable `S3FailureTest` on buildkite?

**Comment by romain-intel** [comment]
Unsurprisingly, the S3FailureTest does *not* pass (nor will it -- we don't use that code path in the new datastore).

File: metaflow/datatools/s3op.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def normalize_client_error(err):
     try:
         return int(error_code)
     except ValueError:
-        if error_code == "AccessDenied":
+        if error_code in ("AccessDenied", "AllAccessDisabled"):
```
Comment (romain-intel) [inline metaflow/datatools/s3op.py:92]: Do we want to do this? This is a change in behavior and technically it seems AllAccessDisabled exists separately from AccessDenied and I guess doesn't mean the same thing.

Comment (oavdeev) [reply metaflow/datatools/s3op.py:92]: Not sure ¯\_(ツ)_/¯ For AWS SDK it is also HTTP code 403 I believe, so it least this line of logic is consistent with that

Comment (romain-intel) [reply metaflow/datatools/s3op.py:92]: I am not opposed to it per say. I actually considered it when the tests started failing but saw that the exception already existed a while ago so decided that if it was already different, there must be a reason (but then again, that's me being rational :) )

**Review by romain-intel — APPROVED** [review]

**Comment by romain-intel** [comment]
OK to merge.

**Pull request merged**